src/Autofac.Integration.Wcf/AutofacInstanceContext.cs

```diff
@@ -103,7 +110,21 @@ public AutofacInstanceContext(ILifetimeScope container)
             {
                 throw new ArgumentNullException("container");
             }
-            this.OperationLifetime = container.BeginLifetimeScope();
+            this.OperationLifetime = !JustInTimeModuleRegistration
+                ? container.BeginLifetimeScope()
```

Comment:
Question: Is the fact that one situation is tagged (with modules) but the other is not (without modules) going to cause people a difficult time in troubleshooting? What if someone intentionally does this somewhere?

```c#
builder.RegisterType<Foo>().InstancePerMatchingLifetimeScope(WcfRequestLifetimeScopeTag);
```

If that's not registered as instance-per-context modules, it'll fail... but if some _totally unrelated_ module _is_ registered instance-per-context, suddenly it'll start working.

I'm curious if tagging the lifetime scope is necessary. Is this to support "instance per lifetime context" objects?

If so, we should tag both when there are registrations _and_ when there aren't.

If it doesn't need to be tagged, let's remove that and the constant altogether.

Reply:
I'll remove the constant altogether